From: Aichun Shi <aichun.shi@intel.com>
Date: Thu, 19 May 2022 16:57:15 +0200
Subject: [PATCH] platform/x86/intel/ifs: fix compile issue
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 0122f5626081382aae00f03c80967436a257d2d7
Modified-by-SEL: No


Fix below compile issue.

"drivers/platform/x86/intel/ifs/load.c: In function ‘ifs_sanity_check’:
drivers/platform/x86/intel/ifs/load.c:193:2: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
  193 |  for (int i = 0; i < total_size / sizeof(u32); i++)
      |  ^~~
drivers/platform/x86/intel/ifs/load.c:193:2: note: use option ‘-std=c99’, ‘-std=gnu99’, ‘-std=c11’ or ‘-std=gnu11’ to compile your code"

Signed-off-by: Aichun Shi <aichun.shi@intel.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/platform/x86/intel/ifs/load.c  | 3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/platform/x86/intel/ifs/load.c b/drivers/platform/x86/intel/ifs/load.c
index d056617ddc85..eed7a65ce1b4 100644
--- a/drivers/platform/x86/intel/ifs/load.c
+++ b/drivers/platform/x86/intel/ifs/load.c
@@ -174,6 +174,7 @@ static int ifs_sanity_check(struct device *dev,
 {
 	unsigned long total_size, data_size;
 	u32 sum, *mc;
+	int i;
 
 	total_size = get_totalsize(mc_header);
 	data_size = get_datasize(mc_header);
@@ -190,7 +191,7 @@ static int ifs_sanity_check(struct device *dev,
 
 	mc = (u32 *)mc_header;
 	sum = 0;
-	for (int i = 0; i < total_size / sizeof(u32); i++)
+	for (i = 0; i < total_size / sizeof(u32); i++)
 		sum += mc[i];
 
 	if (sum) {
-- 
2.35.3

